### DISCUSSION TITLE
can i speed up initialization?

### DISCUSSION THREAD

**Discussion by balfouri1210** [tinymce/tinymce-react#343] [Q&A] [answered]
hi, tinymce fetches **tinymce.min.js** after componentDidMount.

but it takes almost more than 1 second every time.

it's not a big problem, but i think some users can feel bit annoying.

@tiny-james  is there any way to get this faster?

**Comment by tiny-james** [comment, marked as answer]
You can link the TinyMCE script yourself which would allow you to load it on page load which might be a bit faster. Are you using an API key? If you don't specify an API key the load time will be a bit slower.

**Reply by balfouri1210** [reply]
thanks for the reply. yes i'm using API key.
i checked there is `tinymce.min.js` in the npm package.
but tinyMCE download additional javascript file from [https://cdn.tiny.cloud](https://cdn.tiny.cloud).

is this an unavoidable process? bcoz you need to validate API key every time editor loaded?

**Reply by tiny-james** [reply]
The `tinymce.min.js` in the node_modules is used to provide types for typescript - it is not included in the integration at run time.

You can host the editor locally - normally by copying it from the `node_modules` into your `public` folder. There will be docs available describing how to do this some time this week (they are being delayed for unrelated reasons, I've actually had them ready for several weeks now).

Basically the tiny.cloud script is only created when the page does not contain a `tinymce` global already and the `tinymceScriptSrc` was not specified. This is documented here:
https://www.tiny.cloud/docs/tinymce/6/react-ref/#configuring-editor-source

**Reply by tiny-james** [reply]
The docs that I mentioned:
https://www.tiny.cloud/docs/tinymce/6/react-pm-host/

**Reply by balfouri1210** [reply]
thank you so much for great and kind document.
it works perfectly.
